dash/include/dash/matrix/internal/LocalMatrixRef-inl.h

```diff
@@ -445,28 +445,28 @@ ::sub(
 
 template<typename T, dim_t NumDim, dim_t CUR, class PatternT>
 inline LocalMatrixRef<T, NumDim, NumDim-1, PatternT>
-LocalMatrixRef<T, NumDim, CUR, PatternT>
+const LocalMatrixRef<T, NumDim, CUR, PatternT>
```

Comment:
Mhh, the position of the first `const` seems strange to me. Shouldn't a `const` function return a `const` reference? (aka, we need both `const` and non-`const` versions of this function?) 

Also, instead of `inline` maybe these functions can be marked `constexpr`?